core/src/validator.rs

```diff
@@ -201,6 +206,33 @@ impl Validator {
 
         let blocktree = Arc::new(blocktree);
 
+        if config.enable_ctrl_c_handler {
+            let bank_forks = bank_forks.clone();
+            let counter = Arc::new(AtomicUsize::new(0));
+            set_handler(move || {
+                if counter.fetch_add(1, Ordering::Relaxed) > 0 {
+                    warn!("Validator aborting on second ^C");
+                    std::process::exit(1);
+                }
+                let bank = bank_forks.read().unwrap().working_bank();
+                let stake: Option<u64> = bank
+                    .epoch_vote_accounts(bank.get_epoch_and_slot_index(bank.slot()).0)
+                    .and_then(|account| {
+                        account.iter().find(|(pubkey, (stake, account))| {
+                            **pubkey == id
+                                && *stake > 0
+                                && VoteState::deserialize(&account.data).is_ok()
+                        })
+                    })
+                    .map(|(_, (n, _))| *n);
+                match stake {
+                    Some(_val) => warn!("Validator with stake is exiting on first ^C"),
```

Comment:
Let's change this warning to something more more like this:
```rust
warn!("Validator is staked!  Press ^C again to exit"),
```

Reply:
got it